### ISSUE TITLE
String template expression not Working for Union Types

### ISSUE THREAD

**Issue by ThisaruGuruge** [ballerina-platform/ballerina-lang#29674]
**Description:**
Consider the following code:

```ballerina
public type Foo int|float|decimal|string|boolean;

public function main() {
    Foo foo = 4;
    string s = string`${foo}`;
}
```

This results in a compilation error: `incompatible types: expected '(int|float|decimal|string|boolean)', found '(int|float|decimal|string|boolean)'`

But using `${foo.toString()}` works.

**Comment by MaryamZi** [comment]
Related spec issue: https://github.com/ballerina-platform/ballerina-spec/issues/795

**Comment by MaryamZi** [comment]
Fixed in #29733, merged to master in #29934.
